hydrus/data/crud.py

```diff
@@ -210,6 +223,163 @@ def insert(object_: Dict[str, Any], session: scoped_session, id_: Optional[int]
+                        triple = GraphIAC(
+                            subject=instances[index].id,
+                            predicate=property_.id,
+                            object_=class_.id)
+                        triples_list.append(triple)
+
+                    else:
+                        session.close()
+                        raise NotAbstractProperty(type_=prop_name)
+
+                # For insertion in IIT
+                else:
+                    terminal = Terminal(value=objects_[index][prop_name])
+                    session.add(terminal)
+                    session.flush()     # Assigns ID without committing
+
+                    if property_.type_ == "PROPERTY" or property_.type_ == "INSTANCE":
+                        property_.type_ = "INSTANCE"
+                        properties_list.append(property_)
+                        triple = GraphIIT(
+                            subject=instances[index].id,
+                            predicate=property_.id,
+                            object_=terminal.id)
+                        # Add things directly to session, if anything fails
+                        # whole transaction is aborted
+                        triples_list.append(triple)
+                    else:
+                        session.close()
+                        raise NotInstanceProperty(type_=prop_name)
+    session.bulk_save_objects(properties_list)
+    session.bulk_save_objects(triples_list)
+    session.commit()
+    return instance_id_list
+
+
+def update_multiple(ids_: List[int],
+                    type_: str,
+                    objects_: List[Dict[str,
+                                        str]],
+                    session: scoped_session,
+                    api_name: str,
+                    path: str=None) -> List[int]:
+    """
+    To update multiple object using a single request
+    :param ids_: List of ids for objects to be updated
+    :param type_: type of object
+    :param objects_: list of objects
+    :param session: sqlalchemy scoped session
+    :param api_name: api name specified while starting server
+    :param path: endpoint
+    :return: Ids that have been updated
+    """
+    instances = list()
+    ids_string = ids_
+    ids_ = ids_.split(',')
+    for id_ in ids_:
+        instance = get(
+            id_=id_,
+            type_=type_,
+            session=session,
+            api_name=api_name)
+        instance.pop("@id")
+        instances.append(instance)
+        delete(id_=id_, type_=type_, session=session)
+
+    # Try inserting new object
+    try:
+        insert_multiple(objects_=objects_, id_=ids_string, session=session)
+    except (ClassNotFound, InstanceExists, PropertyNotFound) as e:
+        # Put old object back
+        insert_multiple(object_=instances, id_=ids_string, session=session)
+        raise e
+    for id_ in ids_:
+        get(id_=id_, type_=type_, session=session, api_name=api_name, path=path)

```

Comment:
Isn't this loop redundant? If you're returning `ids_` then why do we need this?

Reply:
well yeah it is , ill remove that , and should i remove [this[(https://github.com/httP-APIs/hydrus/blob/develop/hydrus/data/crud.py#L283) too ?

Reply:
Okay, got it! My bad sorry. This is being used to check if objects are inserted successfully. If get fails that'll mean objects were not inserted successfully and exceptions will be raised accordingly.

Reply:
oh okay